draft-kohbrok-mimi-metadata-minimalization.md

```diff
@@ -65,95 +65,227 @@ required to achieve the goals detailed above.
 
 # Pseudonymization
 
-The main mechanism to hide the user and client identifiers from providers in the
-context of individual groups and use user- and client-level pseudonyms instead.
+The main mechanism to hide the user and client identifiers from the hub and
+providers in the context of individual groups and use user- and client-level
+pseudonyms instead.
+
+The main purpose of the pseudonymization scheme described in this document is to
+hide metadata from service providers with as little impact as possible on the
+features provided by the base MIMI protocol. In particular, this means that by
+default, User and client identifiers are still visible to (other) users and
+their clients.
+
+However, the pseudonymization scheme can also be used to hide a user's identity
+from other users. See Section {{user-to-user-pseudonymity}} for more details.
+
+# Pseudonymity through credential encryption
+
+To hide their identities from providers, clients use PseudonymousCredentials to
+represent them in MIMI rooms.
+
+~~~ tls
+struct {
+  IdentifierUri client_pseudonym;
+  IdentifierUri user_pseudonym;
+  opaque signature_public_key;
+  opaque identity_link_ciphertext<V>;
+} PseudonymousCredential
+~~~
+
+As the name suggests, PseudonymousCredentials don't contain a client's client
+and user ID. Instead, they use two pseudonyms: a user pseudonym (shared by all
+other PseudonymousCredentials of the user's clients in a given room) and a
+client pseudonym (unique for each PseudonymousCredential). Each pseudonym is an
+IdentifierUri made up of a randomly generated UUID and the user's provider's
+domain.
+
+~~~ tls
+struct {
+  IdentifierUri client_pseudonym;
+  IdentifierUri user_pseudonym;
+  opaque signature_public_key;
+} PseudonymousCredentialTBS
+
+struct {
+  /* SignWithLabel(., "PseudonymousCredentialTBS",
+    PseudonymousCredentialTBS) */
+  opaque pseudonymous_credential_signature<V>;
+  Credential client_credential;
+} IdentityLinkTBE
+~~~
+
+The `identity_link_ciphertext` is created by encrypting the IdentityLinkTBE,
+which contains the client's real credential, as well as a signature over the
+PseudonymousCredentialTBS using the client credential's signature key.
+
+The identity link key used for encryption is unique per pseudonymous credential. It is derived from the client's connection key.
```

Comment:
I generally agree. However, in this case I'd advocate for using either the name or the identifier since they're pretty much the same.

On a slightly different note, having an "identifier" maybe isn't great in the first place, since it's easily confused with a field name of a struct.